src/cloudformation_cli_python_lib/metrics.py

```diff
@@ -0,0 +1,96 @@
+import logging
+from typing import Any, Mapping
+
+# boto3 doesn't have stub files
+from boto3.session import Session  # type: ignore
+
+from botocore.exceptions import ClientError  # type: ignore
+
+from .interface import Action, MetricTypes, StandardUnit
+
+LOG = logging.getLogger(__name__)
+
+
+def format_dimensions(dimensions: Mapping[Any, Any]) -> Any:  # TODO fix type
+    formatted_dimensions = []
+    for key, value in dimensions.items():
+        formatted_dimensions.append({"Name": key, "Value": value})
+    return formatted_dimensions
+
+
+class MetricPublisher:
+    def __init__(self, namespace: str, session: Session) -> None:
+        self.namespace = namespace
+        self.client = session.client("cloudwatch")
+
+    def _publish_metric(  # pylint: disable-msg=too-many-arguments
+        self,
+        metric_name: MetricTypes,
+        dimensions: Mapping[Any, Any],  # TODO: fix type
+        unit: StandardUnit,
+        value: Any,
+        date: float,
```

Comment:
might be good to call this `timestamp`? i think the type is `datetime.datetime`, [at least the boto3 docs have an example with that](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudwatch.html#CloudWatch.Client.put_metric_data). maybe float works, too? the documentation isn't exactly obvious:

> Timestamp (datetime) -- The time the metric data was received, expressed as the number of milliseconds since Jan 1, 1970 00:00:00 UTC.

also, `value` seems like it should be `float`, not `Any`

Reply:
updated